Modular nvmsgbroker pipeline

chani0343 opened the pull request:
## Task #91: Make nvmsgbroker_pipeline modular for future use

### Changes Made:
- **Refactored `buffer_probe` function** into two separate modular functions:
  - `process_tensor_operations()` - handles tensor processing, softmax normalization, and classification
  - `create_message_metadata()` - creates nvmsgbroker metadata for MQTT transmission
- **Fixed Consumer Dockerfile** - corrected path from `consumer.py` to `database/consumer.py`

### Testing:
✅ **End-to-end testing completed successfully:**
- Pipeline → nvmsgbroker → MQTT → Consumer → MongoDB
- Identical functionality maintained as original monolithic code
- All components working as expected

### Benefits:
- **Modularity**: Functions can be reused in future pipelines
- **Maintainability**: Clear separation of concerns
- **Testability**: Each function can be unit tested separately

### DoD Completed:
✅ Modular probe function working in the same way as before

Closes #91

Commit:
Refactor nvmsgbroker_pipeline to modular functions

Commit:
Merge remote-tracking branch 'origin/main' into modular-nvmsgbroker-pipeline

Commit:
✅ Task #91 Complete: Modular nvmsgbroker_pipeline + Consumer fix

- Refactored buffer_probe into process_tensor_operations and create_message_metadata
- Fixed Consumer Dockerfile path: database/consumer.py
- End-to-end testing successful: Pipeline → nvmsgbroker → MQTT → Consumer → MongoDB
- Ready for code reuse in future pipelines

lyuzinmaxim commented on src/nvmsgbroker_pipeline.py:
```diff
@@ -22,6 +22,53 @@ def load_class_labels():
+        l_user = l_user.next
+    return classification_results
+
+def create_message_metadata(batch_meta, frame_meta, frame_number, classification_results):
```
this function isn't just creating - it also adds new info in-place. Please change the name to represent this

lyuzinmaxim commented on src/nvmsgbroker_pipeline.py:
```diff
@@ -22,6 +22,53 @@ def load_class_labels():
+            probs = np.ctypeslib.as_array(ptr, shape=(83,))
+            exp_probs = np.exp(probs - np.max(probs))
+            normalized_probs = exp_probs / np.sum(exp_probs)
+            top_indices = np.argsort(normalized_probs)[-5:][::-1]
```
this seems to me also as a standalone function

lyuzinmaxim commented on src/nvmsgbroker_pipeline.py:
please add typings

Commit:
Address PR review feedback: improve modularity and documentation

- Added type hints to all functions
- Renamed create_message_metadata to create_and_add_message_metadata
- Extracted apply_softmax_normalization as separate function
- Added docstrings for better code documentation
- Updated documentation with simplified script paths
